projects/npm-tools/packages/npm-scripts/src/scripts/scanExposes.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * SPDX-FileCopyrightText: © 2019 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/* eslint-disable no-console */
+
+const fs = require('fs');
+const path = require('path');
+
+const getMergedConfig = require('../utils/getMergedConfig');
+
+/**
+ * Main script that scans JSP files for JS modules and writes a proposed
+ * `exposes` section for the `npmscripts.config.js` file.
+ */
+module.exports = async function (...args) {
+	const {input} = getMergedConfig('npmscripts', 'build');
+
+	const update = args.includes('-u');
+	const verbose = args.includes('-v');
+
+	const exposes = scanJSPs(input, verbose);
+
+	if (verbose) {
+		console.log('');
+	}
+
+	console.log(exposes);
+
+	if (update) {
+		try {
+			let config = fs.readFileSync('npmscripts.config.js', 'utf8');
+
+			config +=
+				'module.exports.federation = module.exports.federation || {};\n';
+			config += 'module.exports.federation.exposes = [\n';
+
+			exposes.forEach((expose) => {
+				config += `  '${expose}',\n`;
+			});
+
+			config += '];';
+
+			fs.writeFileSync('npmscripts.config.js', config);
+		}
+		catch (err) {
+			if (err.code !== 'ENOENT') {
+				throw err;
+			}
+
+			fs.writeFileSync(
+				'npmscripts.config.js',
+				`/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
```

Comment:
I don't think this is going to work. You have to use the local `copyright.js` file because if this ever runs under `modules/dxp/`, then this is going to be the wrong header (and ESLint will come along and then add the right header _as well_).

Annoying, I know, but it's not going to be too hard to find the right file and read it.

Reply:
This file was intended to be removed. I'll push another commit to do it.